server/opts_test.go

```diff
@@ -2341,7 +2341,11 @@ func TestExpandPath(t *testing.T) {
 			{path: "/Foo/Bar", userProfile: `C:\Foo\Bar`, wantPath: "/Foo/Bar"},
 			{path: "Foo/Bar", userProfile: `C:\Foo\Bar`, wantPath: "Foo/Bar"},
 			{path: "~/Fizz", userProfile: `C:\Foo\Bar`, wantPath: `C:\Foo\Bar\Fizz`},
-			{path: `${HOMEDRIVE}${HOMEPATH}\Fizz`, userProfile: `C:\Foo\Bar`, wantPath: `C:\Foo\Bar\Fizz`},
+			// TODO: (ik), removing this test because there is no "~" so the
```

Comment:
@variadico Can you check here? I am not sure what was the intent of the test:
```
{path: `${HOMEDRIVE}${HOMEPATH}\Fizz`, userProfile: `C:\Foo\Bar`, wantPath: `C:\Foo\Bar\Fizz`},
```
The function expandPath() that you created "works" only if there is a `~` as the first character..

Reply:
Yeah, that's the point of that function. To expand the `~` character to the user's home directory. In that case there is no `~`, but there are env vars that should be expanded.

Did it fail?

Reply:
Yes because HOMEDRIVE and HOMEPATH is not set in that test case, you simply pass userProfile, but again, expandPath() does nothing if it does not start with "~".

Reply:
Aaaah, aah, aah. You're right.

```diff
- {path: `${HOMEDRIVE}${HOMEPATH}\Fizz`, userProfile: `C:\Foo\Bar`, wantPath: `C:\Foo\Bar\Fizz`}
+ {path: `${HOMEDRIVE}${HOMEPATH}\Fizz`, homeDrive: `C:`, homePath: `\Foo\Bar`, wantPath: `C:\Foo\Bar\Fizz`}
```

`os.Expand` should always expand the env vars, even if the path doesn't have a `~`.

https://github.com/nats-io/nats-server/blob/63138509f7a13369271e140ae085a0bd824ee1ef/server/opts.go#L3263

Reply:
Thanks, pushed a fix for the test based on your comment.